### ISSUE TITLE
LinearLrScheduler appears to produce wrong learning rates

### ISSUE THREAD

**Issue by fuzing** [tracel-ai/burn#5124]
I'm using the following learning rate scheduler configuration:

```
    // 1. Define hyperparameter steps and rates
    // total_steps == total number of batches over projected training epochs
    // n_items is the total number of training samples (usize result)
    let total_steps = args.total_projected_epochs * n_items / args.batch_size;

    // warmup_percentage == percentage of total steps used for warmup - 7.5% - 10% of total batches
    let warmup_steps = ((args.total_projected_epochs * n_items / args.batch_size) as f64 * args.warmup_percentage / 100.0).floor() as usize;
    let cosine_steps = total_steps - warmup_steps;

    let max_lr = 1e-3;   // Peak learning rate for Transformer
    let min_lr = 1e-6;   // Final target learning rate

    // 2. Configure the Warmup Phase (Linear increase from min_lr to max_lr)
    let warmup_config = LinearLrSchedulerConfig::new(min_lr, max_lr, warmup_steps);

    // 3. Configure the Decay Phase (Cosine drop from max_lr down to min_lr)
    let cosine_config = CosineAnnealingLrSchedulerConfig::new(max_lr, cosine_steps)
        .with_min_lr(min_lr);

    // 4. Chain them sequentially inside the Composed configuration
    let lr_scheduler = ComposedLrSchedulerConfig::new()
        .linear(warmup_config)
        .cosine(cosine_config)
        .init()?;
```

This configuration produced a "warmup" phase that uses an actual learning rate from much closer to zero (i.e. 1e-9) to a peak of 9.68e-7 (call it 1e-6) and then commences the CosineAnnealing phase from this level, instead of a peak LR of 1e-3 as expected.

Are my expectations mis-aligned with what should happen here, or is this a bug?

<img width="1190" height="928" alt="Image" src="https://github.com/user-attachments/assets/02a8d339-f9cd-4436-95d5-23c75e2ca270" />

I'm using the latest version of burn (main branch), cloned from this repo, as of this morning.

Many thanks!

**Comment by fuzing** [comment]
For additional context, the values of various variables:

```
n_items: 3618345
batch_size: 128
warmup_percentage: 10
Total Steps: 989391
Warmup Steps: 98939
Cosine Steps: 890452
```

**Comment by fuzing** [comment]
Mimicking the linear LR setup and first step from burn-optim/src/lr_scheduler/linear.rs:
```
    // Setup (Line 45 from linear.rs)
    let final_lr = max_lr;
    let step_size = (max_lr - min_lr) / warmup_steps as f64;
    let remaining_iters = warmup_steps + 1;
    eprintln!("final_lr: {final_lr}");
    eprintln!("step_size: {step_size}");
    eprintln!("remaining_iters: {remaining_iters}");

    // First Step (Line 67 from linear.rs)
    let remaining_iters = remaining_iters - (remaining_iters != 0) as usize;
    let lr = final_lr - step_size * remaining_iters as f64;
    eprintln!("remaining_iters: {remaining_iters}");
    eprintln!("lr: {lr}");
```

Yields:
```
Setup:
final_lr: 0.001
step_size: 0.000000010097130555190574
remaining_iters: 98940


After First Step:
remaining_iters: 98939
lr: 0.0000009999999999999159
```

Which appears correct, but in the UI displays as 1e-9 as the initial learning rate instead of the expected lr (0.000000999..., or approximately 1e-6) coming out of the first step.  I'm wondering if the UI is the problem here?

**Comment by fuzing** [comment]
One thought - Perhaps this issue is based on a misconception on my part.   I was presuming that composed schedulers are sequential in nature (i.e. the composition runs one lr scheduler until it hits its target and then the next lr scheduler takes over, etc.), but a further review of the code reveals this is not the case.

Upon further review, it appears they're actually a non-sequential composition based on configurable (Sum|Avg|Prod) reduction of the different schedulers through a mixing function.

But I'm guessing this still raises an important question.  How would one create a multi-stage, "chained" LR-scheduler that, say, increases linearly from 1e-7 to 1e-3 during a warmup phase, then decreases the learning rate (say linearly again) from 1e-3 back to 1e-7.   This doesn't seem easily possible using a Sum, Average, or Product combination, and yet is a common strategy for training all kinds of models.

Is there some other way of achieving this?  If not, perhaps the current method can still work with the introduction of starting/ending iteration constraints such that different LR strategies can be used/combined/excluded during different phases of training.

**Comment by fuzing** [comment]
Pytorch appears to have the following pair of scheduling combinators, namely:

1. Chaining Schedulers:
How it works: Each scheduler adjusts the learning rate obtained by the preceding one. At every step, the new learning rate is the product of all active scaling factors.
Implementation: You pass a list of schedulers to ``torch.optim.lr_scheduler.ChainedScheduler``.
Use case: Applying continuous exponential decay alongside a warm-up phase or a constant learning rate period.

2. Sequential Schedulers
How it works: You define a list of schedulers, and each one takes over the learning rate schedule only when a specific milestone (epoch number) is reached.
Implementation: You pass your list of schedulers and a list of milestone epochs to ``torch.optim.lr_scheduler.SequentialLR``.
Use case: Switching abruptly from a linear warm-up phase to a cosine annealing schedule.

Clearly, burn's analog for ``1`` above is its ``ComposedLrSchedulerConfig``, but there appears to be no analog for ``2``, despite this being a more common use-case.

Note in pytorch the ``milestones`` initializer for transitioning between schedulers (which in Burn's case could be epochs, or some number of iterations which would allow more granular control, but either way would be fine):
```
# The LinearLR will run for the first 5 epochs, then CosineAnnealingLR takes over
combined_scheduler = lr_scheduler.SequentialLR(
    optimizer, 
    schedulers=[warmup_scheduler, cosine_scheduler], 
    milestones=[5]
)
```

**Comment by fuzing** [comment]
Closed in favor of feature request
